src/Server/KeeperReadinessHandler.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_NURAFT
+
+#include <Server/HTTP/HTTPRequestHandler.h>
+#include <Server/HTTP/HTTPRequestHandlerFactory.h>
+#include <Coordination/KeeperDispatcher.h>
+
+namespace DB
+{
+
+class KeeperReadinessHandler : public HTTPRequestHandler, WithContext
+{
+private:
+    std::shared_ptr<KeeperDispatcher> keeper_dispatcher;
+
+public:
+    explicit KeeperReadinessHandler(std::shared_ptr<KeeperDispatcher> keeper_dispatcher_)
+        : keeper_dispatcher(keeper_dispatcher_)
+    {
+    }
+
+    void handleRequest(HTTPServerRequest & request, HTTPServerResponse & response) override;
+};
+
+HTTPRequestHandlerFactoryPtr
+createKeeperHTTPControlMainHandlerFactory(
```

Comment:
Naming is bad here.

Initially I only planned to add a readiness handler, then I thought it would be nice to have an endpoint for yielding leadership, so I renamed the handler to "http control". But it looks like it would be to handle this at SIGTERM rather than in a standalone endpoint.
Also dedicating a port just for "readiness" feels wasteful.